src/network-analytics/azext_network_analytics/_validators.py

```diff
@@ -0,0 +1,39 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from knack.log import get_logger
+import re
+import os
+
+logger = get_logger(__name__)
+
+
+def validate_data_product_name(namespace):
+    name = namespace.data_product_name
+    length = len(name)
+
+    pattern = "^[a-z][a-z0-9]*$"
+    match = re.match(pattern, name)
+    if not match or length < 3 or length > 63:
+        raise InvalidArgumentValueError("Invalid data product name. Please provide a data product name constisting of letter and/or numbers only, ")
```

Comment:
Typo: constisting  --> consisting.
Also the error message isn't accurate because the accepted letters must be lower-case, and the name must start with a letter, and there is the 3-63 length limit....